pandas/io/parquet.py

```diff
@@ -53,9 +53,15 @@ def validate_dataframe(df: DataFrame):
         if not isinstance(df, DataFrame):
             raise ValueError("to_parquet only supports IO with DataFrames")
 
-        # must have value column names (strings only)
-        if df.columns.inferred_type not in {"string", "empty"}:
-            raise ValueError("parquet must have string column names")
+        # must have value column names for all index levels (strings only)
+        if isinstance(df.columns, MultiIndex):
+            if not all(
+                x.inferred_type in {"string", "empty"} for x in df.columns.levels
+            ):
+                raise ValueError("parquet must have string column names")
```

Comment:
can say something about 'for all values in each level of the MultiIndex'

Reply:
Thanks @jreback for the suggestion on the exception statement - adding that into my next commit!